src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -22,66 +25,143 @@ const helpers = require('../lib/helpers');
+
   return h.view('nunjucks/form', {
     ...request.view,
     caption: helpers.getFormTitleCaption(licenceNumber),
     pageTitle: 'Who should the bills go to?',
     back: '/manage',
-    form: sessionForms.get(request, selectCompanyForm(request, company, selectedCompany))
+    form: sessionForms.get(request, selectCompanyForm(request, companies, selectedCompany))
   });
 };
 
 const postCompany = async (request, h) => {
   const { regionId, companyId } = request.params;
-  const { company } = request.pre;
+  const { companies } = request.pre;
   const schema = selectCompanyFormSchema(request.payload);
-  const form = forms.handleRequest(selectCompanyForm(request, company), request, schema);
+  const form = forms.handleRequest(selectCompanyForm(request, companies), request, schema);
+  const { viewData } = await dataService.sessionManager(request, regionId, companyId);
   if (form.isValid) {
     const redirectPath = helpers.processCompanyFormData(request, regionId, companyId, form);
     return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/${redirectPath}`);
   }
-  const { viewData: { redirectPath } } = dataService.sessionManager(request, regionId, companyId);
-  return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId), { redirectPath });
+  return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId), { redirectPath: viewData.redirectPath });
+};
+
+const getSearchCompany = async (request, h) => {
+  const { regionId, companyId } = request.params;
+  // Return the view
+  const { viewData, agent } = dataService.sessionManager(request, regionId, companyId);
+  let selectedContactId = agent ? agent.id : null;
+
+  return h.view('nunjucks/form', {
+    ...request.view,
+    pageTitle: 'Does this contact already exist?',
+    back: `/invoice-accounts/create/${regionId}/${companyId}?redirectPath=${viewData.redirectPath}`,
+    form: sessionForms.get(request, selectExistingCompanyForm(request, selectedContactId))
+  });
+};
+
+const postSearchCompany = async (request, h) => {
+  const { id, filter } = request.payload;
+  const { regionId, companyId } = request.params;
+  const form = forms.handleRequest(selectExistingCompanyForm(request), request, selectExistingCompanySchema);
+  if (form.isValid) {
+    if (id === 'new') {
+      const volatileKey = uuid();
+      const path = `/contact-entry/new?` + queryString.stringify({
+        searchQuery: filter,
+        sessionKey: volatileKey,
+        back: `/invoice-accounts/create/${regionId}/${companyId}/contact-search`,
+        redirectPath: `/invoice-accounts/create/${regionId}/${companyId}/contact-entry-complete?sessionKey=${volatileKey}`
+      });
+      return h.redirect(path);
+    } else {
+      // Set the display company name
+      await dataService.sessionManager(request, regionId, companyId, { agent: { companyId: id } });
```

Comment:
You need to add the agent company name here to the view data

Reply:
You're totally right. My apologies.